include/util/struct_packing.hpp

```diff
@@ -0,0 +1,19 @@
+#ifndef STRUCT_PACKING_HPP
+#define STRUCT_PACKING_HPP
+
+/*
+    Macro to pack struct/class in both environment
+    Usage:  PACK( 
+        struct MyStruct 
+        { 
+            int Property1;
+        });
+*/
+
+#ifdef _MSC_VER
+#define PACK( input ) __pragma( pack(push, 1) ) input __pragma( pack(pop) )
```

Comment:
gcc supports pragma(pack), so it is safe to use with gcc  4.4.7, icc 13.0, clang 3.0 and all  above version.